docs/docs/dynamo-store/architecture.md

```diff
@@ -0,0 +1,117 @@
+
+With these constraints enumerated it should be clear why an event-per-row
+schema is not the greatest idea
+
+1. Reading a stream necessitates a scan in which you pay 0.5RUs at minimum for
+   each event in the stream
+2. Writing will always be transactional so you'll pay double
+
+## The architecture
+
+At heart DynamoDB is a document store and an event sourced system is composed
+of multiple streams. It therefore stands to reason that we should model the
+storage such that the stream is the document. This works great until we run into
+the 400KB max item size. To solve for this we will move the current events to
+another document once the document size reaches a configurable threshold (in
+events or bytes). Streams are generally short-lived so calving should be a rare
+occurrence. For streams where it's unavoidable we provide functionality to store
+a snapshot alongside the main stream document ensuring that the state of the
+stream can be reconstructed in at most 1 round trip.
+
+To summarise:
+
+- A stream is represented as one or more documents in the store (a Tip document
+  and 0 or more calved batches of events)
+- The Tip document can contain snapshots (we call these unfolds)
+- Since attribute keys count towards item-size we use single letter key names
+
+## Table schema
+
+| Key         | Type                                | Description                                                    |
+| ----------- | ----------------------------------- | -------------------------------------------------------------- |
+| `p`artition | String                              | The stream name (hash key)                                     |
+| `i`ndex     | Number                              | Int32.MAX for `Tip` 0-N for calved batches (range key)         |
+| `b`ytes     | ?Number                             | Only exists in the Tip, number of bytes in predecessor batches |
+| `etag`      | ?String                             | Etag of latest update (used for OCC)                           |
+| `e`vents    | List(Map([Event](#event-schema)))   | [Events](#event-schema)                                        |
+| `c`ases     | List(String)                        | Event types                                                    |
+| `u`nfolds   | List(Map([Unfold](#unfold-schema))) | [Unfolds](#unfold-schema)                                      |
+| `a`ppends   | Number                              | The number of events appended in the latest update             |
+| `n`umber    | Number                              | The version of the stream                                      |
+
+<h2 id="event-schema">Event schema</h2>
+
+| Key             | Type    | Description                                                  |
+| --------------- | ------- | ------------------------------------------------------------ |
+| `t`imestamp     | String  | ISO8601 DateTime representing the time the event was written |
```

Comment:
when it was encoded in the app - better not to oversimplify
Not sure if you saw but I made it DateTimeOffset.UtcNow recently
might eb worth it to convey that it potentially includes/stores the offset?

Reply:
Yep, will fix the time of the time 😅.

ISO8601 should convey that offsets may be in play just fine, no?